src/corppass/corppass-helper-ndi.ts

```diff
@@ -190,15 +197,22 @@ export class NdiOidcHelper {
 			const {
 				data: { keys },
 			} = await this.axiosClient.get<{ keys: Object[] }>(finalJwksUri, { headers: this.additionalHeaders });
-			const jwsVerifyKey = JSON.stringify(keys[0]);
 
 			const { id_token } = tokens;
 
 			const finalDecryptionKey = overrideDecryptKey ?? this.jweDecryptKey;
 			const decryptedJwe = await JweUtil.decryptJWE(id_token, finalDecryptionKey.key, finalDecryptionKey.format);
 			const jwsPayload = decryptedJwe.payload.toString();
-			const verifiedJws = await JweUtil.verifyJWS(jwsPayload, jwsVerifyKey, "json");
-			return JSON.parse(verifiedJws.payload.toString()) as NDIIdTokenPayload;
+			let error = null;
+			for (const key of keys) {
+				try {
+					const verified = await JweUtil.verifyJWS(jwsPayload, JSON.stringify(key), "json");
+					return JSON.parse(verified.payload.toString()) as NDIIdTokenPayload;
+				} catch (err) {
+					error = err;
+				}
+			}
+			throw error;
```

Comment:
Is it safe to simply `throw error;`?
Or should we be doing a check on whether there's an error before throwing?